website/docs/style-theme.mdx

```diff
@@ -33,5 +34,81 @@ Navigation.setRoot({
 });
 ```
 
+## Conditional themes with Options Processors
+
+Some style requirements can't be facilitated with the `defaultOptions` api. For example, an app may have a unique theme for screens displayed in a modal or require that modals have a default dismiss button in the TopBar. Options Processors alow us to mutate the options object of each screen and layout right before they are displayed.
+
+Lets see how we can leverage this API to add a dismiss button to the TopBar when showing a modal.
+
+```ts
+Navigation.addOptionProcessor<OptionsTopBar>(
+  'topBar',
+  (topBar: OptionsTopBar, commandName: CommandName): OptionsTopBar => {
+    if (commandName === CommandName.ShowModal) {
+      if (!topBar.leftButtons) {
+        topBar.leftButtons = [
+          {
+            id: 'dismissModalButton',
+            icon: require('dismissIcon.png'),
+            color: 'black',
+          },
+        ];
+      }
+    }
+    return topBar;
+  }
+);
+
+// Now that each modal has a dismiss button, let's handle the button press event and dismiss the modal when needed.
+Navigation.events().registerNavigationButtonPressedListener(
+  (event: NavigationButtonPressedEvent) => {
+    if (event.buttonId === 'dismissModalButton') {
+      Navigation.dismissModal(event.componentId);
+    }
+  }
+);
+```
+
+Options Processors can also be used to set default options for buttons, here's how:
+
+```ts
+Navigation.addOptionProcessor<OptionsTopBarButton>(
+  'topBar.rightButtons',
+  (button: OptionsTopBarButton, commandName: CommandName): OptionsTopBarButton => {
+    button.fontFamily = 'helvetica';
+    button.color = 'red';
+    return button;
+  }
+);
+```
+
+## Conditional themes with Layout Processors
+
+A Layout Processor is similar in concept to the Options Processor discussed above. While Options Processors are invoked for each options object, a Layout Processor is invoked once on the entire (layout)['../api/layout-layout/'] tree passed to the Navigation command.
```

Comment:
Should `A Layout Processor` be written as `Layout Processors` to be consistent with `Options Processors`?

`Options Processor` -> `Options Processors`